### ISSUE TITLE
UserError: ACC007 -- JAR does not contain a manifest

### ISSUE THREAD

**Issue by glassfishrobot** [javaee/glassfish#21215]
background:

Java Web Start
Java Web Start allows your application client to be easily launched and automatically downloaded and updated. It is enabled for all application clients by default. For more information, see Using Java Web Start.

GlassFish Server Open Source Edition
Application Development Guide Release 4.0
page 172

and

Downloading and Launching an Application Client
If Java Web Start is enabled for your deployed application client, you can launch it for testing. Simply click on the Launch button next to the application client or application's listing on the App Client Modules page in the Administration Console.

GlassFish Server Open Source Edition
Application Development Guide Release 4.0
page 176

This, in fact, is incorrect. Appclient applications which run from the IDE launch from Glassfish missing the manifest. Glassfish fails to include the manifest with the JNLP file, resulting in errors such as:

org.glassfish.appclient.client.acc.UserError: ACC007: The app client file
/___JWSappclient/___app/SingletonQueue/SingletonQueueClient/
SingletonACCClient.jar does not contain a manifest; the app client
container cannot process it. Embedded programs should pass URIs with
scheme "jar:" for JAR files and scheme "file:" for directories.
#### Environment
ubuntu 14.04
#### Affected Versions
[4.1]

**Comment by vinayvishal** [comment]
There are not enough details available in the bug description. Its pretty much possible that app itself is working fine and the client is missing some configurations causing it to fail. The bug doesn't have any sample app attached so that the issue can be reproduced.
With what I have tried, the ear gets deployed successfully. The client is retrieved during deployment and then is successfully being launched.
This is the  error encountered by the filer:
```bash
ACC007.diag.cause.1 = The file might not be a valid app client JAR or undeployed EAR. It might be another kind of file or have become corrupted.
```
This error might be a genuine one and indeed there could be some issue with the way client jar was built or if client jar is trying to access the remote component differently.

Its not clear from bug description if problem is just with java web start or its happening even with standalone client too?

This is what I tried and it worked for me:
```bash
vinay-20170713-224153$ asadmin deploy --retrieve . ear/target/helloworld.ear 
Application deployed with name helloworld.
Command deploy executed successfully.
```
This deployed the ear and retrieved the client jar `helloworldClient.jar` in the directory specified.
on invoking client, the expected output was returned:
```bash
vinay-20170714-104736$ appclient -jar helloworldClient.jar
My Bean going to say hello
Appclient saying Hello World!
```
Since there is not enough steps/ information or a sample app available, we will look at it in a future release.
